lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -3135,6 +3158,273 @@ int s1ap_handle_paging_request(
+          .e_rab_not_to_be_modified_list.item[index]
+          .s1_xNB_fteid.teid = htonl(*((uint32_t*) erab_item->dL_GTP_TEID.buf));
+
+      /** Set the IP address from the FTEID. */
+      if (4 == blength(transport_layer_address)) {
+        S1AP_E_RAB_MODIFICATION_IND(message_p)
+            .e_rab_not_to_be_modified_list.item[index]
+            .s1_xNB_fteid.ipv4 = 1;
+        memcpy(
+            &S1AP_E_RAB_MODIFICATION_IND(message_p)
+                 .e_rab_not_to_be_modified_list.item[index]
+                 .s1_xNB_fteid.ipv4_address,
+            transport_layer_address->data, blength(transport_layer_address));
+      } else if (16 == blength(transport_layer_address)) {
+        S1AP_E_RAB_MODIFICATION_IND(message_p)
+            .e_rab_not_to_be_modified_list.item[index]
+            .s1_xNB_fteid.ipv6 = 1;
+        memcpy(
+            &S1AP_E_RAB_MODIFICATION_IND(message_p)
+                 .e_rab_not_to_be_modified_list.item[index]
+                 .s1_xNB_fteid.ipv6_address,
+            transport_layer_address->data, blength(transport_layer_address));
+      } else {
+        AssertFatal(
+            0, "TODO IP address %d bytes", blength(transport_layer_address));
+      }
+      bdestroy_wrapper(&transport_layer_address);
+
+      S1AP_E_RAB_MODIFICATION_IND(message_p)
+          .e_rab_not_to_be_modified_list.no_of_items++;
+    }
+  }
+  rc = send_msg_to_task(&s1ap_task_zmq_ctx, TASK_MME_APP, message_p);
+  OAILOG_FUNC_RETURN(LOG_S1AP, rc);
+}
+
+//------------------------------------------------------------------------------
+void s1ap_mme_generate_erab_modification_confirm(
+    s1ap_state_t* state, const itti_s1ap_e_rab_modification_cnf_t* const conf) {
+  uint8_t* buffer_p        = NULL;
+  uint32_t length          = 0;
+  ue_description_t* ue_ref = NULL;
+  S1ap_S1AP_PDU_t pdu      = {0};
+  S1ap_E_RABModificationConfirm_t* out;
+  S1ap_E_RABModificationConfirmIEs_t* ie = NULL;
+
+  OAILOG_FUNC_IN(LOG_S1AP);
+  DevAssert(conf != NULL);
+
+  if ((ue_ref = s1ap_state_get_ue_mmeid(conf->mme_ue_s1ap_id)) == NULL) {
+    OAILOG_ERROR(
+        LOG_S1AP,
+        "This mme ue s1ap id (" MME_UE_S1AP_ID_FMT
+        ") is not attached to any UE context\n",
+        conf->mme_ue_s1ap_id);
+    OAILOG_FUNC_OUT(LOG_S1AP);
+  }
+
+  memset(&pdu, 0, sizeof(pdu));
+  pdu.present = S1ap_S1AP_PDU_PR_successfulOutcome;
+  pdu.choice.successfulOutcome.procedureCode =
+      S1ap_ProcedureCode_id_E_RABModificationIndication;
+  pdu.choice.successfulOutcome.criticality = S1ap_Criticality_reject;
+  pdu.choice.successfulOutcome.value.present =
+      S1ap_SuccessfulOutcome__value_PR_E_RABModificationConfirm;
+  out = &pdu.choice.successfulOutcome.value.choice.E_RABModificationConfirm;
+
+  /* mandatory */
+  ie = (S1ap_E_RABModificationConfirmIEs_t*) calloc(
+      1, sizeof(S1ap_E_RABModificationConfirmIEs_t));
+  ie->id            = S1ap_ProtocolIE_ID_id_MME_UE_S1AP_ID;
+  ie->criticality   = S1ap_Criticality_ignore;
```

Comment:
Should the criticality be reject for this Mandatory IE?

Reply:
Criticality is Ignore according to 3GPP-36.413 v15.3.0